pilot/proxy/envoy/config.go

```diff
@@ -568,7 +568,7 @@ func buildDestinationHTTPRoutes(sidecar model.Node, service *model.Service,
 
 		if useDefaultRoute {
 			// default route for the destination is always the lowest priority route
-			cluster := buildOutboundCluster(service.Hostname, servicePort, nil)
+			cluster := buildOutboundCluster(service.Hostname, servicePort, nil, service.External())
```

Comment:
can you just pass the service object itself, so that you can do the External() check inside the buildoutboundcluster

Reply:
Note that in the case of https://github.com/istio/istio/pull/2730/files/db07de4c3037e474595b757b5e32da12ac8b140f#diff-da6fd41d1e67c4875373b9f50e4e29a3L348, you do not have a service, but rule's mirror/destination:
```go
fqdnDest := model.ResolveHostname(config.ConfigMeta, rule.Mirror)
```

Another example:
```go
actualDestination = model.ResolveHostname(config.ConfigMeta, dst.Destination)
```